addons/website_slides/static/src/js/slides_course_join.js

```diff
@@ -1,28 +1,59 @@
 odoo.define('website_slides.course.join.widget', function (require) {
 'use strict';
 
-var core = require('web.core');
+var core = require('web.core');  
 var publicWidget = require('web.public.widget');
-require('website_slides.slides');
 
 var _t = core._t;
 
 var CourseJoinWidget = publicWidget.Widget.extend({
-    template: 'slide.course.join',
+    template: 'slide.quiz.join',
     xmlDependencies: ['/website_slides/static/src/xml/channel_management.xml'],
     events: {
         'click .o_wslides_js_course_join_link': '_onClickJoin',
     },
-
-    init: function (parent, channelId){
-        this.channelId = channelId;
-        return this._super.apply(this, arguments);
+    
+    init: function (parent, options) {
+        this._super.apply(this, arguments);
+        this.quizWidget = parent || false;
+        this.quiz = options.quiz || false;
+        this.signupAllowed = options.channel.signupAllowed || false;
+        this.channelId = options.channel.channelId || false;
+        this.channelEnroll = options.channel.channelEnroll || false;
+        this.isMember = options.isMember || false;
+        this.publicUser = options.publicUser || false;
+        this.slide = options.slide || false;
```

Comment:
I did it like this because when CourseJoinWidget is used in websiteSlidesCourseJoin options only is:
`var options = {channel: {channelEnroll: data.channelEnroll, channelId: data.channelId}};`
but when  CourseJoinWidget is used in Quiz options is:

```
            var options = {
                quiz: this.quiz,
                channel: this.channel,
                isMember: this.isMember,
                publicUser: this.publicUser,
                slide: this.slide,                
            };
```
So only channelId and channelEnroll are present in both cases.

Reply:
That seems like a lot of additional values to pass.
That widget probably doesn't need to get the "quiz", but I let you get to my other previous comments to see why.

The rest of the options do not seem related to a quiz usage as well.
So maybe you should pass all values even if "websiteSlidesCourseJoin" ?
Not sure what behavior they imply in the join Widget.

I'll check again when you're done with other comments, I'll be able to see more clearly then.

Reply:
I think it is better now :)